tensorrec/tensorrec.py

```diff
@@ -282,13 +284,23 @@ def _build_tf_graph(self, n_user_features, n_item_features):
         tf_interactions_serial = tf_interactions.values
         self.tf_rankings = rank_predictions(tf_prediction=self.tf_prediction)
 
-        # Loss function nodes
-        self.tf_basic_loss = self.loss_graph_factory(tf_prediction_serial=self.tf_prediction_serial,
-                                                     tf_interactions_serial=tf_interactions_serial,
-                                                     tf_prediction=self.tf_prediction,
-                                                     tf_interactions=tf_interactions,
-                                                     tf_rankings=self.tf_rankings,
-                                                     tf_alignment=self.tf_alignment)
+        # Compose loss function args
+        # This composition is for execution safety: it prevents loss functions that are incorrectly configured from
+        # having visibility of certain nodes.
+        loss_graph_kwargs = {
+            'tf_prediction_serial': self.tf_prediction_serial,
+            'tf_interactions_serial': tf_interactions_serial,
```

Comment:
Is it on purpose that one of the values above calls self and the other doesn't?

Reply:
Yes -- the ones with `self.` are graph hooks that the TensorRec object keeps track of. The rest are not used outside of the graph construction method.